src/qt/walletmodel.cpp

```diff
@@ -187,8 +187,7 @@ WalletModel::SendCoinsReturn WalletModel::prepareTransaction(WalletModelTransact
             setAddress.insert(rcp.address);
             ++nAddresses;
 
-            CScript scriptPubKey = GetScriptForDestination(DecodeDestination(rcp.address.toStdString()));
-            CRecipient recipient = {scriptPubKey, rcp.amount, rcp.fSubtractFeeFromAmount};
+            CRecipient recipient{DecodeDestination(rcp.address.toStdString()), rcp.amount, rcp.fSubtractFeeFromAmount};
```

Comment:
To check my understanding:

The bug was introduced by _not_ updating this line in #28246 to use the `CTxDestination` returned by `DecodeDestination` directly in the `CRecipient` object. Instead, it was converting to a `CScript` and silently converting to a `CNoDestination`, since `CRecipient` now expects a `CTxDestination`. So if we had marked the `CNoDestination` constructor as explicit, the compiler would have warned us here.

I changed it back to try and use the `CScript` directly in the `CRecipient` and confirmed I got a compiler warning:

```
qt/walletmodel.cpp:192:34: error: could not convert ‘spk’ from ‘const CScript’ to ‘CTxDestination’
```